### ISSUE TITLE
Git command hang up freezing UI 

### ISSUE THREAD

**Issue by ipod825** [glepnir/galaxyline.nvim#167]
https://github.com/glepnir/galaxyline.nvim/blob/d544cb9d0b56f6ef271db3b4c3cf19ef665940d5/lua/galaxyline/provider_vcs.lua#L105

Is `--contains` necessary in the command "git branch -a --no-abbrev --contains"? I find it's blocking and freeze the whole neovim ui when there are hundreds of thousands of commits.

**Comment by Shatur95** [comment]
It's a really bad idea to call `git` command just to get branch. Statusline hangs a lot for because of it. Lualine, for example, [just parses the file](https://github.com/hoob3rt/lualine.nvim/blob/master/lua/lualine/components/branch.lua) that contains branches.

**Comment by weeman1337** [comment]
Yeah. Depending on the case it takes ages and burns the CPU. The detached head detection should somehow be optimised. Additionally is it possible to change it to run asynchronously?

Until this issue has been solved you could use the following (if you use [fugitive](https://github.com/tpope/vim-fugitive)):
```
...
provider = function () return vim.fn.FugitiveHead() end,
...
```
